app/controllers/nonsipp/memberdetails/CheckMemberDetailsFileController.scala

```diff
@@ -45,22 +48,30 @@ class CheckMemberDetailsFileController @Inject()(
   formProvider: YesNoPageFormProvider,
   uploadService: UploadService,
   uploadValidator: MemberDetailsUploadValidator,
+  schemeDateService: SchemeDateService,
+  auditService: AuditService,
   val controllerComponents: MessagesControllerComponents,
   view: YesNoPageView
 )(implicit ec: ExecutionContext, mat: Materializer)
-    extends FrontendBaseController
+    extends PSRController
     with I18nSupport {
 
   private val form = CheckMemberDetailsFileController.form(formProvider)
 
   def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn).async { implicit request =>
+    val startTime = System.currentTimeMillis
     val preparedForm = request.userAnswers.fillForm(CheckMemberDetailsFilePage(srn), form)
     val uploadKey = UploadKey.fromRequest(srn)
 
     uploadService.getUploadStatus(uploadKey).map {
       case None => Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
-      case Some(upload: UploadStatus.Success) =>
+      case Some(upload: UploadStatus.Success) => {
+        audit(srn, upload, startTime)
         Ok(view(preparedForm, viewModel(srn, Some(upload.name), mode)))
+      }
+      case Some(failure: UploadStatus.Failed) =>
```

Comment:
UploadStatus.Failed needed to be changed in order to distinguish this pattern from `Some(_)`